tensorflow/python/framework/func_graph.py

```diff
@@ -347,7 +347,7 @@ def control_dependencies(self, control_inputs):
     for c in control_inputs:
       # Check for _UnreadVariable
       if (isinstance(c, ops.IndexedSlices) or
-          (hasattr(c, "_handle") and hasattr(c, "op"))):
+          (resource_variable_ops.is_resource_variable(c) and hasattr(c, "op"))):
```

Comment:
Just using the `hasattr(c, "_handle")` check would fail for `AutoCastDistributedVariable`.